packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -41,6 +41,57 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
   @override
   ContainerLayer get parent => super.parent;
 
+  // Whether this layer has any changes since its last call to [addToScene].
+  //
+  // Initialized to true as a new layer has never called [addToScene].
+  bool _needsAddToScene = true;
+
+  /// Mark that this layer has changed and [addToScene] needs to be called.
+  @protected
+  void markNeedsAddToScene() => _needsAddToScene = true;
```

Comment:
trivial nit: use a block for this, otherwise you're returning bool despite being declared void

Reply:
Done.